infra/modules/database/main.tf

```diff
@@ -0,0 +1,362 @@
+  engine_version             = aws_rds_cluster.db.engine_version
+  auto_minor_version_upgrade = true
+  monitoring_role_arn        = aws_iam_role.rds_enhanced_monitoring.arn
+  monitoring_interval        = 30
+}
+
+resource "random_password" "random_db_password" {
+  length = 48
+  # Remove '@' sign from allowed characters since only printable ASCII characters besides '/', '@', '"', ' ' may be used.
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "aws_ssm_parameter" "random_db_password" {
+  name  = "/db/${var.name}/master-password"
+  type  = "SecureString"
+  value = random_password.random_db_password.result
+}
+
+resource "aws_kms_key" "db" {
+  description         = "Key for RDS cluster ${var.name}"
+  enable_key_rotation = true
+}
+
+#-----------------------#
+# Network Configuration #
+#-----------------------#
+
+resource "aws_security_group" "db" {
+  name_prefix = "${var.name}-db"
+  description = "Database layer security group"
+  vpc_id      = var.vpc_id
+}
+
+resource "aws_security_group" "role_manager" {
+  name_prefix = "${var.name}-role-manager"
+  description = "Database role manager security group"
+  vpc_id      = var.vpc_id
+}
+
+resource "aws_vpc_security_group_egress_rule" "role_manager_egress_to_db" {
+  security_group_id = aws_security_group.role_manager.id
+  description       = "Allow role manager to access database"
+
+  from_port                    = 5432
+  to_port                      = 5432
+  ip_protocol                  = "tcp"
+  referenced_security_group_id = aws_security_group.db.id
+}
+
+resource "aws_vpc_security_group_ingress_rule" "db_ingress_from_role_manager" {
+  security_group_id = aws_security_group.db.id
+  description       = "Allow inbound requests to database from role manager"
+
+  from_port                    = 5432
+  to_port                      = 5432
+  ip_protocol                  = "tcp"
+  referenced_security_group_id = aws_security_group.role_manager.id
+}
+
+#----------------#
+# Authentication #
+#----------------#
+
+resource "aws_iam_policy" "db_access" {
+  name   = "${var.name}-db-access"
+  policy = data.aws_iam_policy_document.db_access.json
+}
+
+data "aws_iam_policy_document" "db_access" {
+  # Policy to allow connection to RDS via IAM database authentication
+  # which is more secure than traditional username/password authentication
+  # https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.IAMPolicy.html
+  statement {
+    actions = [
+      "rds-db:connect"
+    ]
+
+    resources = [
+      "${local.db_user_arn_prefix}/${local.app_username}",
+      "${local.db_user_arn_prefix}/${local.migrator_username}",
+    ]
+  }
+}
+
+#------------------#
+# Database Backups #
+#------------------#
+
+# Backup plan that defines when and how to backup and which backup vault to store backups in
+# See https://docs.aws.amazon.com/aws-backup/latest/devguide/about-backup-plans.html
+resource "aws_backup_plan" "backup_plan" {
```

Comment:
I had never heard of this service. Curious if we have experience with it. How does it compare to https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster#backup_retention_period

I think that backup_retention_period gives you the restore to point in time feature, enabling restores to any second in the period, up to 35 days.

Reply:
Started a 🔒 [slack thread](https://nava.slack.com/archives/C205Q1RA8/p1686150185045439) to discuss this with @aplybeah and @rocketnova since this was based off of some of their work.

Reply:
I'll merge for now but am interested/open to simplifying this if we can get by with just backup_retention_period instead of backup plan